src/game.js

```diff
@@ -0,0 +1,46 @@
+function game() {
+
+}
+
+LAST_FRAME = 10;
+
+game.prototype.frameScoreWithBonus = function(frames, selectedFrameNumber) {
+  if(selectedFrameNumber === LAST_FRAME) return this.tenthFrameScoring(frames, selectedFrameNumber);
+  if(frames[selectedFrameNumber -1].isAStrike()) return this.strikeScoring(frames, selectedFrameNumber);
+  if(frames[selectedFrameNumber -1].isASpare()) return this.spareScoring(frames, selectedFrameNumber);
+  return this.basicScoring(frames, selectedFrameNumber);
+};
+
+game.prototype.tenthFrameScoring = function(frames, selectedFrameNumber) {
+  return frames[selectedFrameNumber - 1].preBonusScore();
+};
+
+game.prototype.strikeScoring = function(frames, selectedFrameNumber) {
+  finalFrameScore = 0;
+  if(selectedFrameNumber === frames.length - 1) {
+    finalFrameScore = frames[selectedFrameNumber - 1].preBonusScore() + frames[selectedFrameNumber].firstScore() + frames[selectedFrameNumber].secondScore();
+  }
+  else {
+    finalFrameScore = frames[selectedFrameNumber - 1].preBonusScore() + this.frameScoreWithBonus(frames, selectedFrameNumber + 1);
+  }
+  if(finalFrameScore > 30) {
+    finalFrameScore = 30;
```

Comment:
It feels like we are silently swallowing an error case here - the final frame score should never be more than 30 and if it is then we should throw an exception because that indicates a bug somewhere in the game logic.